core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1112,7 +1079,8 @@ class UnifiedLog(@volatile var logStartOffset: Long,
 
     new LogAppendInfo(firstOffset, lastOffset, lastLeaderEpochOpt, maxTimestamp, offsetOfMaxTimestamp,
       RecordBatch.NO_TIMESTAMP, logStartOffset, RecordConversionStats.EMPTY, sourceCompression, targetCompression,
-      shallowMessageCount, validBytesCount, monotonic, lastOffsetOfFirstBatch)
+      shallowMessageCount, validBytesCount, monotonic, lastOffsetOfFirstBatch, Collections.emptyList[RecordError], null,
+      LeaderHwChange.NONE)
```

Comment:
Why do we need to explicitly set default value here?

Reply:
Earlier, we had default values in Scala. These changes did not introduce constructor overloading for different combinations as the usages were low.  It is simpler to have a constructor and set this value from a couple of places.